core/tauri/src/plugin.rs

```diff
@@ -52,6 +53,145 @@ pub trait Plugin<R: Runtime>: Send {
   fn extend_api(&mut self, invoke: Invoke<R>) {}
 }
 
+type SetupHook<R> = dyn Fn(&AppHandle<R>) -> Result<()> + Send + Sync;
+type OnWebviewReady<R> = dyn Fn(Window<R>) + Send + Sync;
+type OnEvent<R> = dyn Fn(&AppHandle<R>, &Event) + Send + Sync;
+
+/// Builds a tauri plugin
+pub struct Builder<R: Runtime> {
+  name: &'static str,
+  invoke_handler: Box<InvokeHandler<R>>,
+  setup: Box<SetupHook<R>>,
+  initialization_script: Option<String>,
+  on_page_load: Box<OnPageLoad<R>>,
+  on_webview_ready: Box<OnWebviewReady<R>>,
+  on_event: Box<OnEvent<R>>,
+}
+
+impl<R: Runtime> Builder<R> {
+  /// Creates a new Plugin builder.
+  pub fn new(name: &'static str) -> Self {
+    Self {
+      name,
+      setup: Box::new(|_| Ok(())),
+      initialization_script: None,
+      invoke_handler: Box::new(|_| ()),
+      on_page_load: Box::new(|_, _| ()),
+      on_webview_ready: Box::new(|_| ()),
+      on_event: Box::new(|_, _| ()),
+    }
+  }
+
+  /// Defines the JS message handler callback.
+  pub fn invoke_handler<F>(mut self, invoke_handler: F) -> Self
+  where
+    F: Fn(Invoke<R>) + Send + Sync + 'static,
+  {
+    self.invoke_handler = Box::new(invoke_handler);
+    self
+  }
+
+  /// The JS script to evaluate on webview initialization.
+  /// The script is wrapped into its own context with `(function () { /* your script here */ })();`,
+  /// so global variables must be assigned to `window` instead of implicity declared.
+  ///
+  /// It's guaranteed that this script is executed before the page is loaded.
+  pub fn initialization_script(mut self, initialization_script: String) -> Self {
```

Comment:
What do you think about `js_init_script`?

Reply:
I like it

Reply:
Changed to `js_init_script`!